src/main/resources/jupyter/cluster-site.conf

```diff
@@ -9,12 +9,30 @@
     SSLVerifyClient require
     SSLVerifyDepth 10
 
+    RewriteEngine on
+
+    # RStudio
+    RewriteCond %{HTTP:Upgrade} =websocket
+    RewriteCond %{REQUEST_URI} /notebooks/[^/]*/[^/]*/rstudio/.* [NC]
+    RewriteRule /notebooks/[^/]*/[^/]*/rstudio/(.*)     ws://127.0.0.1:8001/$1  [P,L]
+
+    RewriteCond %{HTTP:Upgrade} !=websocket
+    RewriteCond %{REQUEST_URI} /notebooks/[^/]*/[^/]*/rstudio/.* [NC]
+    RewriteRule /notebooks/[^/]*/[^/]*/rstudio/(.*)     http://127.0.0.1:8001/$1 [P,L]
+
+    ProxyPass /notebooks/[^/]*/[^/]*/rstudio/ http://127.0.0.1:8001/
```

Comment:
This isn't ideal because it works on `/notebooks/project/cluster/rstudio/` but not `/notebooks/project/cluster/rstudio`. But maybe phase 3 can address this.

Reply:
Also if these apache configs become too much of a headache we could consider shifting some of the rewrite logic into the Scala proxy code. I kind of like the logic localized in 1 place through, and the Scala proxy being a pass-through.

Reply:
how will this work when we support BYOT? won't we need to move it down to the actual app in some manner at that point anyway?

this is forward-thinking and not a blocker but wanted to raise the Q since it felt relevant to your second comment

Reply:
That's a good point. Yeah editing this apache conf won't be scalable for BYOT, we'd probably have to move logic in the app.